src/test/java/bio/terra/FileTypeDetectorTest.java

```diff
@@ -16,7 +16,7 @@
   public void tarGzipHandlerTest() {
     try {
       StorageObjectData storageObjectData =
-          GsonWrapper.getInstance().fromJson(MOCK_EVENT_GZIP, StorageObjectData.class);
+          GsonWrapper.convertFromClass(MOCK_EVENT_GZIP, StorageObjectData.class);
       FileTypeDetector fileTypeDetector = new FileTypeDetector(storageObjectData);
       // Set a mock TGZ input stream to simulate the GCS input stream
       fileTypeDetector.setInputStream(MOCK_TGZ);
```

Comment:
This test is invalid. It sets the input stream, then reads the data stream, without ever calling any of the `handle*` methods. Therefore, when it reads the data stream, there's nothing there. 

The catch (Exception e) {} silently catches and ignores any errors. Please change the catch clause to something like:
```
    } catch (Exception e) {
      fail("test failed with error: " + e.getMessage());
    }
```

and, the `while ((archiveEntry = ais.getNextEntry()) != null) {` loop only executes any assertions if it actually finds anything in `getNextEntry()`. This test should have some kind of assertion that it actually processed the expected number of archive entries.

Reply:
Added missing `handleMediaType()` step.